### ISSUE TITLE
[firebase_messaging] Management of data messages when activity is killed

### ISSUE THREAD

**Issue by kroikie** [FirebaseExtended/flutterfire#829]
If the app is in a killed state and a data message is sent, the app will call the onMessageReceived of the FlutterFirebaseMessagingService, but since the Activity is killed the FirebaseMessagingPlugin can not respond to the intent and the message is lost.

This is a problem when somebody wants to perform an action depending on the data which is contained in a message.

Are there any plans to handle this kind of behaviour?

**Comment by kroikie** [comment]
@jankuss

The issue at https://github.com/flutter/flutter/issues/17482 has been closed and moved here. Future collaboration on this issue will be done here.

**Comment by iapicca** [comment]
Hi @jankuss
If you are still experiencing this issue 
with the latest version of Flutter and FlutterFire plugin
can you please provide your updated  `flutter doctor -v`
and your and your `flutter run --verbose`/`flutter build --verbose` ?
Also, to better address the issue, would be helpful
if you could post a self contained app on github
or the steps to reproduce it.
Thank you

https://github.com/FirebaseExtended/flutterfire/pull/53
https://github.com/FirebaseExtended/flutterfire/issues/1055
https://github.com/FirebaseExtended/flutterfire/issues/766
https://github.com/FirebaseExtended/flutterfire/issues/508

**Comment by iapicca** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. We are therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact descriptions what happens, logs and the output of 'flutter doctor -v'  please.
All system setups can be slightly different so its always better to open new issues and reference related issues.

**Comment by creativecreatorormaybenot** [comment]
@iapicca I am afraid I do not understand why you would close this. This is a feature request rather than a bug report.

**Comment by kroikie** [comment]
@iapicca I believe that this will be resolved once the migration of the firebase_messaging plugin to use the v2 embedding for Android. I'll reopen this and then close once that migration is complete.

**Comment by creativecreatorormaybenot** [comment]
@kroikie Is this currently being worked on or what PR do I need to follow?

**Comment by kroikie** [comment]
@creativecreatorormaybenot no PR yet but it is being worked on, sorry for the delay here.

**Comment by cmaster11** [comment]
> @creativecreatorormaybenot no PR yet but it is being worked on, sorry for the delay here.

@kroikie , do you have a reference to an issue/milestone/anything related to this migration?

**Comment by Salakar** [comment]
Hey all 👋 

As part of our roadmap (#2582) we've just shipped a complete rework of the `firebase_messaging` plugin that aims to solve this and many other issues. As far as I can tell this is now supported by using the `onBackgroundMessage` API which is not tied to an Activity.

If you can, please try out the dev release (see the [migration guide](https://firebase.flutter.dev/docs/migration/) for upgrading and for changes) and if you have any feedback then join in the discussion [here](https://github.com/FirebaseExtended/flutterfire/discussions/4023).

Given the scope of the rework I'm going to go ahead and close this issue in favor of trying out the latest plugin.

Thanks everyone.
